From 39e570976c3480963d498ad78817deb845e76d69 Mon Sep 17 00:00:00 2001
From: Yongxin Dai <yodai@qnx.com>
Date: Fri, 14 Nov 2025 11:29:18 -0500
Subject: [PATCH 2/2] fix missing function getrandom


 100.0% lib/
diff --git a/lib/ephy-sync-utils.c b/lib/ephy-sync-utils.c
index 75c08fc..b15e3bf 100644
--- a/lib/ephy-sync-utils.c
+++ b/lib/ephy-sync-utils.c
@@ -36,6 +36,10 @@
 #include <unistd.h>
 #endif
 
+#if defined(__QNX__)
+#include <fcntl.h>
+#endif
+
 static const char hex_digits[] = "0123456789abcdef";
 
 const SecretSchema *
@@ -167,6 +171,19 @@ ephy_sync_utils_base64_urlsafe_decode (const char *text,
   return out;
 }
 
+#if defined(__QNX__)
+static ssize_t getrandom(void *buf, size_t buflen, unsigned int flags)
+{
+  ssize_t ret;
+  int fd = open("/dev/urandom", O_RDONLY);
+  if (fd < 0)
+    return -1;
+  ret = read(fd, buf, buflen);
+  close(fd);
+  return ret;
+}
+#endif
+
 /*
  * This is mainly required by Nettle's RSA support.
  * From Nettle's documentation: random_ctx and random is a randomness generator.
-- 
2.51.2

